src/react/inject.jsx

```diff
@@ -39,7 +52,16 @@ export default injectFunction => WrappedComponent => class extends Component {
     const { injectedProps } = this.state
     const newInjectedProps = injectFunction(this.context.store, nextProps || this.props)
 
-    if (shallowEqual(injectedProps, newInjectedProps)) return
+    if (
```

Comment:
you may should use `shouldComponentUpdate` hook to do it ?

With something like that : 
```es6
shouldComponentUpdate(nextProps) {
    const { injectedProps } = this.state
    const newInjectedProps = injectFunction(this.context.store, nextProps)
    return 
        nextProps !== undefined 
        && !shallowEqual(this.props, nextProps)
        || !shallowEqual(withoutFunctions(injectedProps), withoutFunctions(newInjectedProps))
}

inject = (nextProps) => {
    const newInjectedProps = injectFunction(this.context.store, nextProps || this.props)
    this.setState(state => ({ ...state, injectedProps: newInjectedProps, }))
}
```

Reply:
I have to write it in `inject` because `inject` is called by the store subscriber :)

Reply:
`inject` will mutate the state then you can stop the update of the component with `shouldComponentUpdate`:
```
shouldComponentUpdate(nextProps, nextState)
```